src/disc/disc.py

```diff
@@ -11,6 +11,44 @@
 ADMINISTRATOR = 1 << 3
 VIEW_CHANNEL = 1 << 10
 
+class User:
+    def __init__(self, id: str, username: str, discriminator: str, global_name: str=None, avatar: str=None) -> None:
+        self.id = id
+        self.username = username
+        self.discriminator = discriminator
+        self.global_name = global_name
+        self.avatar = avatar
+    
+    def __str__(self) -> str:
+        return f"{self.username}, {self.id}"
+    
+    def __repr__(self) -> str:
+        return f"{self.username}, {self.id}"
+
+class GuildMember:
+    def __init__(self, user: User = None, roles: list[str] = None) -> None:
+        self.user = user
+        self.roles = roles
+    
+    def __str__(self) -> str:
+        return f"{self.user.id}, {self.user.username}, {str(self.roles)}"
+
+    def __repr__(self) -> str:
+        return f"{self.user.username}"
+    
+class Channel:
+    def __init__(self, id: str, type: int, guild_id: str, name: str, permission_overwrites: list[str] = None) -> None:
+        self.id = id
+        self.type = type
+        self.name = name
+        self.permission_overwrites = permission_overwrites
+
+    def __str__self(self) -> str:
+        return f"{self.id}, {self.name}"
+    
+    def __repr__(self) -> str:
+        return f"{self.id}, {self.name}"
+
```

Comment:
Let's update these to dataclasses, so we can be explicit about typing! https://docs.python.org/3.11/library/dataclasses.html

Reply:
Done.